libewf/libewf_single_files.c

```diff
@@ -988,50 +1533,2314 @@ int libewf_single_files_parse_record_values(
+					sizeof(uint8_t) * value_string_size);
+
+				if (single_file_source->manufacturer == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_INSUFFICIENT,
+						"%s: unable to create manufacturer.",
+						function);
+
+					goto on_error;
+				}
+				if (narrow_string_copy(
+					single_file_source->manufacturer,
+					value_string,
+					value_string_size - 1) == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_SET_FAILED,
+						"%s: unable to set manufacturer.",
+						function);
+
+					goto on_error;
+				}
+				single_file_source->manufacturer[value_string_size - 1] = 0;
+
+				single_file_source->manufacturer_size = value_string_size;
+			}
+			else if ((type_string[0] == (uint8_t) 'l')
+				  && (type_string[1] == (uint8_t) 'o')
+				  && (type_string[2] == (uint8_t) 'c'))
+			{ 
+				single_file_source->location = (uint8_t *)memory_allocate(
+					sizeof(uint8_t) * value_string_size);
+
+				if (single_file_source->location == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_INSUFFICIENT,
+						"%s: unable to create location.",
+						function);
+
+					goto on_error;
+				}
+				if (narrow_string_copy(
+					single_file_source->location,
+					value_string,
+					value_string_size - 1) == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_SET_FAILED,
+						"%s: unable to set location.",
+						function);
+
+					goto on_error;
+				}
+				single_file_source->location[value_string_size - 1] = 0;
+
+				single_file_source->location_size = value_string_size;
+			}
+		}
+		else if (type_string_size == 3)
+		{
+			if ((type_string[0] == (uint8_t) 'i')
+			 && (type_string[1] == (uint8_t) 'd'))
+			{
+				if (libfvalue_utf8_string_copy_to_integer(
+					value_string,
+					value_string_size,
+					&value_64bit,
+					32,
+					LIBFVALUE_INTEGER_FORMAT_TYPE_DECIMAL_UNSIGNED,
+					error) != 1)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_SET_FAILED,
+						"%s: unable to set id.",
+						function);
+
+					goto on_error;
+				}
+				single_file_source->id = (uint32_t)value_64bit;
+			}
+			else if ((type_string[0] == (uint8_t) 'g')
+				  && (type_string[1] == (uint8_t) 'u'))
+			{
+				single_file_source->guid = (uint8_t *)memory_allocate(
+					sizeof(uint8_t) * value_string_size);
+
+				if (single_file_source->guid == NULL)
+				{
+					libcerror_error_set(
+						error,
+						LIBCERROR_ERROR_DOMAIN_MEMORY,
+						LIBCERROR_MEMORY_ERROR_INSUFFICIENT,
+						"%s: unable to create GUID.",
+						function);
+
+					goto on_error;
+				}
+
+				if (libewf_single_files_read_hex_string(
+					value_string,
+					value_string_size,
+					single_file_source->guid,
```

Comment:
encase refers to this a device guid

Reply:
Yeah - it would be clearer to refer to it as such rather than as a more generic "guid"